project-koku/koku
# fix:  compare integers not integer strings

## Description

This change will fix a code-smell in one of our comparisons in the `filter_s3_objects_less_than` function.

In this comparison `if metadata_value < metadata_value_check` ,  we compare string integers like ("4" < "6") which could be problematic in case one of the variables is an invalid integer string as well as cause unexpected results as show below:
```py
>>> "100" < "1"
False

...

>>> "100" < None
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: '<' not supported between instances of 'str' and 'NoneType'
>>> 

```

## Testing

1. Checkout Branch
2. Restart Koku
3. Load test data
4. See that data ingestion and processing completes successfully

## Notes

...
